Src/DependencyCollector/Shared.Tests/Implementation/SqlClientDiagnosticSourceListenerTests.cs

```diff
@@ -0,0 +1,434 @@
+            {
+                OperationId = operationId,
+                Operation = "ExecuteReader",
+                ConnectionId = sqlConnection.ClientConnectionId,
+                Command = sqlCommand,
+                Exception = new Exception("Boom!"),
+                Timestamp = 2000000L
+            };
+
+            var now = DateTimeOffset.UtcNow;
+
+            this.fakeSqlClientDiagnosticSource.Write(
+                SqlClientDiagnosticSourceListener.SqlErrorExecuteCommand,
+                commandErrorEventData);
+
+            var dependencyTelemetry = (DependencyTelemetry)this.sendItems.Single();
+
+            Assert.AreEqual(commandErrorEventData.Exception.ToInvariantString(), dependencyTelemetry.Properties["Exception"]);
+            Assert.IsFalse(dependencyTelemetry.Success.Value);
+        }
+        
+        [TestMethod]
+        public void TracksConnectionOpenedError()
+        {
+            var operationId = Guid.NewGuid();
+            var sqlConnection = new SqlConnection(TestConnectionString);
+            
+            var beforeOpenEventData = new
+            {
+                OperationId = operationId,
+                Timestamp = 1000000L
+            };
+
+            this.fakeSqlClientDiagnosticSource.Write(
+                SqlClientDiagnosticSourceListener.SqlBeforeOpenConnection,
+                beforeOpenEventData);
+
+            var errorOpenEventData = new
+            {
+                OperationId = operationId,
+                Operation = "Open",
+                ConnectionId = sqlConnection.ClientConnectionId,
+                Connection = sqlConnection,
+                Exception = new Exception("Boom!"),
+                Timestamp = 2000000L
+            };
+
+            var now = DateTimeOffset.UtcNow;
+
+            this.fakeSqlClientDiagnosticSource.Write(
+                SqlClientDiagnosticSourceListener.SqlErrorOpenConnection,
+                errorOpenEventData);
+
+            var dependencyTelemetry = (DependencyTelemetry)this.sendItems.Single();
+
+            Assert.AreEqual(errorOpenEventData.OperationId.ToString("N"), dependencyTelemetry.Id);
+            Assert.AreEqual(errorOpenEventData.Operation, dependencyTelemetry.Data);
+            Assert.AreEqual("(localdb)\\MSSQLLocalDB | master | " + errorOpenEventData.Operation, dependencyTelemetry.Name);
+            Assert.AreEqual("(localdb)\\MSSQLLocalDB | master", dependencyTelemetry.Target);
+            Assert.AreEqual(RemoteDependencyConstants.SQL, dependencyTelemetry.Type);
+            Assert.AreEqual(1000000L, dependencyTelemetry.Duration.Ticks);
+            Assert.IsTrue(dependencyTelemetry.Timestamp.Ticks < now.Ticks);
+            Assert.AreEqual(errorOpenEventData.Exception.ToInvariantString(), dependencyTelemetry.Properties["Exception"]);
+            Assert.IsFalse(dependencyTelemetry.Success.Value);
+        }
+
+        [TestMethod]
+        public void TracksConnectionCloseError()
```

Comment:
Close() is not to be tracked, so this can be removed

Reply:
Done.

Reply:
Done.